c10/core/impl/cow/state_machine.h

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include <c10/core/impl/cow/shadow_storage.h>
+#include <c10/macros/Macros.h>
+#include <c10/util/Optional.h>
+#include <c10/util/intrusive_ptr.h>
+
+#include <cstdint>
+#include <mutex>
+
+namespace c10::impl::cow {
+
+// Responsible for managing the copy-on-write simulation state
+// machine.
```

Comment:
but what is the cow simulation state machine 🤔

Reply:
hopefully the expanded comment covers this